### ISSUE TITLE
[MPH-13] Show _ALL_ plugins and settings in help:effective-pom

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-help-plugin#177]
**[Kenney Westerhof](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kenney)** opened **[MPH-13](https://issues.apache.org/jira/browse/MPH-13?redirect=false)** and commented

Currently running 'mvn help:effective-pom' only shows the resource plugin and the help plugin in the
build section.

Running 'mvn compile help:effective-pom' also shows the compiler plugin.

I'd like to see ALL plugins that are used in the lifecycle to show up in the output, not only those that
are configured.

Also, I'd like to see the default settings for plugins in there. For instance, on a jdk 1.4 project,
where the compiler plugin has not been configured, I'd like to see the source/target versions explicitly set.
Only if you manually specify a jdk 1.5 project by setting source/target to 5.0 in the pom, then it shows.

In short: currently only POM defaults are listed. Expand this by listing all plugins and plugin defaults.
(and explicit settings, ofcourse).

---

3 votes, 0 watchers

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

This is not a bug in the Help plugin, but in Maven. Using Maven 3.0-alpha-7 yields the results you want.
